tsconfig.json

```diff
@@ -2,7 +2,8 @@
   "extends": "@tsconfig/node16/tsconfig.json",
   "files": [],
   "compilerOptions": {
-    "noEmit": true
+    "noEmit": true,
+    "allowJs": true
```

Comment:
was wondering if this is needed? since the types still rebuild fine without it

Reply:
slightly more unrelated to this pr, noticed

```json
{
  "extends": "@tsconfig/node16/tsconfig.json",
  "files": []
}
```

is all we need for types to still rebuild successfully

so can we get rid of the remaining clutter? 

---

again could get this merged first and i could follow-up

Reply:
Yeah, I can't remember what I was thinking regarding `noEmit`.  It seemed like a good idea to default to "don't emit anything", but given it'd only apply to files which aren't part of the build _anyway_, there's no point.

As mentioned here and above, these config files affect more than just the build, and this root one serves as sort of "default settings" for the entire repo.  It also comes into play when using `@typescript-eslint/parser` (and Wallaby, but that's a tool only I use, afaik); there are files that are linted with this parser but are otherwise not part of the build.  

There's also some interplay with VSCode settings such as "implicit type-checking in JavaScript files". 

I'm not going to be able to verify that `@typescript-eslint/parser` actually needs that `allowJs: true` in _this_ PR (will need to do that in #753), but using a sane default like `@tsconfig/node16` seems reasonable, at minimum.

Are you saying that we can remove the references and `tsc` will find all of the `tsconfig.json` files in the repo and build them??

Reply:
> Yeah, I can't remember what I was thinking regarding `noEmit`. It seemed like a good idea to default to "don't emit anything", but given it'd only apply to files which aren't part of the build _anyway_, there's no point.
> 
> As I mentioned above, these config files affect more than just the build, and this root one serves as sort of "default settings" for the entire repo. It also comes into play when using `@typescript-eslint/parser` (and Wallaby, but that's a tool only I use, afaik); there are files that are linted with this parser but are otherwise not part of the build.
> 
> There's also some interplay with VSCode settings such as "implicit type-checking in JavaScript files".
> 
> I'm not going to be able to verify that `@typescript-eslint/parser` actually needs that `allowJs: true` in _this_ PR (will need to do that in #753), but using a sane default like `@tsconfig/node16` seems reasonable, at minimum.

👍 i bought Quokka with Wallaby discounted in their bundle (love a good deal), but barely touched Wallaby 😅

> Are you saying that we can remove the references and `tsc` will find all of the `tsconfig.json` files in the repo and build them??

i tested removing the references then rebuilding types, but not fully inspecting the outputs are what/how we want oops, after some more reading can see why they're needed